docs/proposals/17-shoot-control-plane-migration-bad-case.md

```diff
@@ -0,0 +1,102 @@
+
+However, there might be situations in which `gardenlet` in the source seed cluster can't connect to the Garden cluster and determine that `spec.seedName` has changed. Similarly, the connection to the seed `kube-apiserver` could also be broken. This might be caused by issues with the seed cluster itself. In other situations, the migration flow steps in the source seed might have started but might not be able to finish successfully. In all such cases, it should still be possible to migrate a shoot's control plane to a different seed, even though executing the migration flow steps in the source seed might not be possible. The potential "split brain" situation caused by having the shoot's control plane components attempting to reconcile the shoot resources in two different seeds must still be avoided, by ensuring that the shoot's control plane in the source seed is deactivated before it is activated in the destination seed.
+
+The mechanisms and adaptations described below have been tested as part of a PoC prior to describing them here.
+
+## Owner Election / Copying Snapshots
+
+To achieve the goals outlined above, an "owner election" (or rather, "ownership passing") mechanism is introduced to ensure that the source and destination seeds are able to successfully negotiate a single "owner" during the migration. This mechanism is based on special *owner DNS records* that uniquely identify the seed that currently hosts the shoot's control plane ("owns" the shoot).
+
+For example, for a shoot named `i500152-gcp` in project `dev` that uses an internal domain suffix `internal.dev.k8s.ondemand.com` and is scheduled on a seed with an identity `shoot--i500152--gcp2-0841c87f-8db9-4d04-a603-35570da6341f-sap-landscape-dev`, the owner DNS record is a TXT record with a domain name `owner.i500152-gcp.dev.internal.dev.k8s.ondemand.com` and a single value `shoot--i500152--gcp2-0841c87f-8db9-4d04-a603-35570da6341f-sap-landscape-dev`. The owner DNS record is created and maintained by reconciling an `owner` DNSRecord resource, if the recently introduced DNSRecords feature is enabled via the `UseDNSRecords` feature gate. 
+
+Unlike other extension resources, the `owner` DNSRecord resource is not reconciled every time the shoot is reconciled, but only when the resource is created. Therefore, the owner DNS record value (the owner ID) is updated only when the shoot is migrated to a different seed. For more information, see [Add handling of owner DNSRecord resources](https://github.com/gardener/gardener/pull/4307).
+
+The owner DNS record domain name and owner ID are passed to components that need to perform ownership checks, such as the `backup-restore` container of the `etcd-main` StatefulSet, and all extension controllers. These components then check regularly whether the actual owner ID (the value of the record) matches the passed ID. If they don't, the ownership check is considered failed, which causes the special behavior described below.
```

Comment:
What if the components in the source seed aren't able to resolve DNS names (maybe because the underlying infrastructure has networking issues)?

Reply:
I added a new section "Add Handling Inability to Resolve the Owner DNS Record" with the above explanation, it captures what we already discussed.